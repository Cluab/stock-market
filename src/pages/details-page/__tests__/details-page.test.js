import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { MemoryRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import DetailsPage from '../details-page';
import '@testing-library/jest-dom/extend-expect';

const middleware = [thunk];
const initialState = [[{
  symbol: 'AAPL',
  date: '2022-09-24',
  revenue: 394328000000,
  costOfRevenue: 223546000000,
  costAndExpenses: 274891000000,
  ebitda: 133138000000,
  grossProfit: 170782000000,
  grossProfitRatio: 0.43309630561360085,
  operatingExpenses: 51345000000,
  operatingIncome: 119437000000,
  eps: 6.15,
  operatingIncomeRatio: 0.30288744395528594,
}]];
const mockStore = configureMockStore(middleware);
const store = mockStore({ stocks: initialState });
describe('CompaniesPage component', () => {
  test('should match with snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <DetailsPage test="AAPL" />
          </Router>
        </Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the Header and CompaniesList components', () => {
    const { getByTestId } = render(
      <Provider store={store}><Router><DetailsPage test="AAPL" /></Router></Provider>,
    );
    expect(getByTestId('header')).toBeInTheDocument();
    expect(getByTestId('Details')).toBeInTheDocument();
  });
});
