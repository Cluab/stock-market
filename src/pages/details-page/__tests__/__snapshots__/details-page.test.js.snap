// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesPage component should match with snapshot 1`] = `
Array [
  <header
    className="header"
    data-testid="header"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa-solid fa-gear"
      />
    </button>
    <h1>
      stock market
    </h1>
    <a
      href="/"
      onClick={[Function]}
    >
      <span
        className="fa-solid fa-chevron-left test"
      />
    </a>
  </header>,
  <div
    className="main-comp-section"
    data-testid="Details"
  >
    <span
      className="fa-solid fa-money-bill-trend-up"
    />
    <div
      className="main-comp-name"
    >
      <h2>
        AAPL
      </h2>
      <span>
        394328000000
        / Revenue
      </span>
    </div>
  </div>,
  <div
    className="main-page-bar"
  >
    AAPL
      BREAKDOWN  -  
    2022-09-24
  </div>,
  <div>
    <ul
      className="comp-details"
    >
      <li>
        <p>
          revenue
        </p>
         
        <span
          data-testid="test-revenue"
        >
          394328000000
        </span>
      </li>
      <li>
        <p>
          costOfRevenue
        </p>
         
        <span>
          223546000000
        </span>
      </li>
      <li>
        <p>
          costAndExpenses
        </p>
         
        <span>
          274891000000
        </span>
      </li>
      <li>
        <p>
          ebitda
        </p>
         
        <span>
          133138000000
        </span>
      </li>
      <li>
        <p>
          grossProfit
        </p>
         
        <span>
          170782000000
        </span>
      </li>
      <li>
        <p>
          grossProfitRatio
        </p>
         
        <span>
          0.43309630561360085
        </span>
      </li>
      <li>
        <p>
          operatingExpenses
        </p>
         
        <span>
          51345000000
        </span>
      </li>
      <li>
        <p>
          operatingIncome
        </p>
         
        <span>
          119437000000
        </span>
      </li>
      <li>
        <p>
          Earnings per share
        </p>
         
        <span
          data-testid="test-eps"
        >
          6.15
        </span>
      </li>
      <li>
        <p>
          OIR
        </p>
         
        <span>
          0.30288744395528594
        </span>
      </li>
    </ul>
  </div>,
]
`;
